hudi-common/src/main/java/org/apache/hudi/common/model/WriteOperationType.java

```diff
@@ -48,6 +48,8 @@
   INSERT_OVERWRITE_TABLE("insert_overwrite_table"),
   // compact
   COMPACT("compact"),
+  // z-order
```

Comment:
Do we need another write type for z-order? For my first think, `z-order` is a layout optimize for the table which is much similar to `order-by`. As currently we already have the `clustering`  to  optimize the layout by `order-by`, can we implement the z-order by under the `clustering` ? So that we can support two kind of data clustering:

> Clustering table order by col1;
> Clustering table zorder by col2;

Reply:
we support two operation type
1. cluster  support
2. zorder optimize by overwrite table。

Reply:
It make sense to me now.

Reply:
you mean you sql?   good suggest,  now  we only support api .   i will add sql support for anther pr.   thanks for your suggest

Reply:
No, I mean if **1. cluster support** can cover the function of **2. zorder optimize by overwrite table**, we can only implement the zorder by under the clustering. No need to introduce a write type  for z-order. As you know, hudi currently has too many write types, which add the cost of use for our users.

Reply:
@pengzhiwei2018    Due to the implementation of Hudi itself, optimze by overwritetable is more convenient to use， since cluster operation need to set many properties。   
Another consideration: as we know delta lake support z-order,  just like delta lake api,  we want keep the same implement

Reply:
I'm +1 to keep this as a clustering strategy. If needed, we can make top level API different. But underneath, WriteClient API and ActionExecutor will benefit from using same code (I do see lot of copy paste code there right now). Any changes we do in one ActionExecutor will have to be copied to other which is going to be a challenge.